src/MongoDB.EntityFrameworkCore/Extensions/MongoEntityTypeBuilderExtensions.cs

```diff
@@ -115,4 +116,77 @@ public static class MongoEntityTypeBuilderExtensions
 
         return entityTypeBuilder.CanSetAnnotation(MongoAnnotationNames.CollectionName, name, fromDataAnnotation);
     }
+
+    /// <summary>
+    /// Configures the element name that the entity is mapped to when stored as an embedded document.
+    /// </summary>
+    /// <param name="entityTypeBuilder">The builder for the entity type being configured.</param>
+    /// <param name="name">The name of the parent element.</param>
+    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
+    public static OwnedNavigationBuilder SetElementName(
+        this OwnedNavigationBuilder entityTypeBuilder,
+        string? name)
+    {
+        entityTypeBuilder.OwnedEntityType.SetContainingElementName(name);
+
+        return entityTypeBuilder;
+    }
+
+    /// <summary>
+    /// Configures the element name that the entity is mapped to when stored as an embedded document.
+    /// </summary>
+    /// <param name="entityTypeBuilder">The builder for the entity type being configured.</param>
+    /// <param name="name">The name of the parent element.</param>
+    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
+    public static OwnedNavigationBuilder<TOwnerEntity, TDependentEntity> SetElementName<TOwnerEntity, TDependentEntity>(
+        this OwnedNavigationBuilder<TOwnerEntity, TDependentEntity> entityTypeBuilder,
+        string? name)
+        where TOwnerEntity : class
+        where TDependentEntity : class
+    {
+        entityTypeBuilder.OwnedEntityType.SetContainingElementName(name);
+
+        return entityTypeBuilder;
+    }
+
+    /// <summary>
+    /// Configures the element name that the entity is mapped to when stored as an embedded document.
+    /// </summary>
+    /// <param name="entityTypeBuilder">The builder for the entity type being configured.</param>
+    /// <param name="name">The name of the parent element.</param>
+    /// <param name="fromDataAnnotation">Indicates whether the configuration was specified using a data annotation.</param>
+    /// <returns>The same builder instance if the configuration was applied, <see langword="null" /> otherwise.</returns>
+    public static IConventionEntityTypeBuilder? SetElementName(
+        this IConventionEntityTypeBuilder entityTypeBuilder,
+        string? name,
+        bool fromDataAnnotation = false)
+    {
+        if (!entityTypeBuilder.CanSetElementName(name, fromDataAnnotation))
+        {
+            return null;
+        }
```

Comment:
It does seem odd but that's what the Cosmos one does https://github.com/dotnet/efcore/blob/release/8.0/src/EFCore.Cosmos/Extensions/CosmosPropertyBuilderExtensions.cs#L80 - frankly I'd have expected it to throw...

Reply:
`IConvention*Builder` types are not Fluent API and returning `null` is expected in cases where an overriding configuration is present.

Reply:
It might be, but from user prospective: this is extension method to IConventionPropertyBuilder which returns IConventionPropertyBuilder... it looks like I can chain more configuration code (that's why I see this as fluent). And if I chain more code, it might result to NullReferenceException, instead of some more reasonable exception. So we might want to either to change it to be void or throw there for consistency.

Reply:
Whenever one of these methods returns `null` further configuration should be avoided, so the calls should be chained using `?.` instead of `.`